### PR TITLE
Reduce HTTP calls logging from background workers that are just polling data from chain

### PR DESCRIPTION
**Pull request by antonyshchenko**
The idea is to use thread-local variable to silence HTTP client logs in busy background worker threads that are just polling data from the chain in a loop

### CODE REVIEW

File: backend/src/main/kotlin/xyz/funkybit/core/blockchain/bitcoin/JsonRpcClientBase.kt [hunk 1]
```diff
@@ -50,20 +49,19 @@ open class JsonRpcClientBase(val url: String, interceptor: Interceptor? = null)
...
         }
         .build()
 
     open val mediaType = "text/plain".toMediaTypeOrNull()
 
-    fun call(requestString: String, logResponseBody: Boolean = true, noLog: Boolean = false): JsonElement {
+    fun call(requestString: String): JsonElement {
         val body = requestString.toRequestBody(mediaType)
 
-        if (!noLog) {
```
Comment (antonyshchenko) [inline backend/src/main/kotlin/xyz/funkybit/core/blockchain/bitcoin/JsonRpcClientBase.kt:64]: Replaced this logging with HttpClient interceptor

File: src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (antonyshchenko) [inline src/main/resources/log4j2.xml:1]: This logging configuration is only used when running locally

File: backend/src/main/kotlin/xyz/funkybit/core/repeater/Repeater.kt [hunk 3]
```diff
@@ -32,7 +33,7 @@ class Repeater(db: Database, private val automaticTaskScheduling: Boolean = true
         ProgramUtxoRefresherTask(),
     ).associateBy { it.name }
 
-    private val timer = ScheduledThreadPoolExecutor(8)
+    private val timer = ScheduledThreadPoolExecutor(8, BasicThreadFactory.Builder().namingPattern("repeater-pool-%d").build())
```
Comment (antonyshchenko) [inline backend/src/main/kotlin/xyz/funkybit/core/repeater/Repeater.kt:36]: Better thread names in the logs

**Review by bflood366 — APPROVED** [review]
Looks great 👍 , should save us a lot of time

File: backend/src/main/kotlin/xyz/funkybit/core/utils/HttpClient.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            requestBuffer.readUtf8()
+        } ?: ""
+
+        val response = chain.proceed(request)
+        val contentType: MediaType? = response.body?.contentType()
+        val responseBody = response.body?.string()
+
+        logger.debug { "HTTP call: url=${request.url}, request=${abbreviateString(requestBodyCopy)}, response code=${response.code}, response body=${responseBody?.let(::abbreviateString)}" }
+
+        // making a copy of response body since it can be consumed only once
```
Comment (bholzman) [inline backend/src/main/kotlin/xyz/funkybit/core/utils/HttpClient.kt:50]: Nit: remove comment

**Review by bholzman — APPROVED** [review]
Looks great!

File: backend/src/main/kotlin/xyz/funkybit/core/utils/HttpClient.kt (hunk 4 above)
Comment (antonyshchenko) [reply backend/src/main/kotlin/xyz/funkybit/core/utils/HttpClient.kt]: done

**Pull request merged**